pointers/pointers.c

```diff
@@ -50,6 +61,39 @@ char *find_char(char *str, int c)
 */
 char *find_string(char *haystack, char *needle)
 {
+    // initialize a new pointer to the needle pointer
+    char *beginN = needle;
+    int found = 0;
+    // while the deref haystack does not equal null char, increment
+    while (*haystack != '\0')
+    {
+        // if deref haystack equals deref needle
+        if (*haystack == *needle && found == 0)
+        {
+            // initialize new pointer to at haystack pointer
+            char *beginH = haystack;
+            // while value of beginH and beginN are equal and deref beginN does not equal null char
+            while (*beginH == *beginN)
+            {
+                // increment beginN pointer
+                beginN++;
+                // increment beginH pointer
+                beginH++;
+                // if beginN equals null char
+                if (*beginN == '\0')
+                {
+                    // change found to 1
+                    found = 1;
+                    // return haystack
+                    return haystack;
+                }
+            }
```

Comment:
Can your program handle the following:  
haystack = "I need a need", 
needle = "needle"
?  (i.e. what happens when beginH ='\0' before beginN='\0'?)